prov/rxm/src/rxm_domain.c

```diff
@@ -589,6 +597,20 @@ static int rxm_mr_regattr(struct fid *fid, const struct fi_mr_attr *attr,
 	rxm_mr->device = msg_attr.device.reserved;
 	*mr = &rxm_mr->mr_fid;
 
+	// get the GDR copy registration if possible, inspired from
+	// efa_mr_hmem_setup
+	if (cuda_is_gdrcopy_enabled()) {
```

Comment:
Controlling the behavior of non-cuda iface with a cuda only method does not look right. Can we have an iface-aware method here?

Reply:
that's an issue I have quickly discussed last week with @shijin-aws, I can try to add a cuda agnostic function but I would expect i to be another PR.
FWIW, the issue is shared with EFA (https://github.com/ofiwg/libfabric/blob/14594abdcf3b664eece27e4978efd1ae82497a94/prov/efa/src/efa_mr.c#L260).

Reply:
I merged https://github.com/ofiwg/libfabric/pull/9456. 

I think we should introduce a new ops named like `is_dev_reg_enabled`

Reply:
To be fair, EFA's code ^^ is executed within a dedicated function for cuda iface (for `iface == FI_HMEM_CUDA`) so there is no risk here.

Reply:
I have started here: https://github.com/ofiwg/libfabric/pull/9469
but I am not familiar enough with the different backend (rocm etc) to finish it

Reply:
I think it is safe to leave it as false for the iface that you are not familiar. We are just introducing a new op and it's not necessary to make all providers to onboard it.